app/models/grda_warehouse/hud/client.rb

```diff
@@ -1852,6 +1858,8 @@ def self.strict_search(criteria, client_scope: none)
           map { |m| m.destination_client.id }
         where(id: ids)
       rescue RangeError
+        # FIXME: what is this range error?
+        Sentry.capture_exception(e)
```

Comment:
I was able to hit this by searching for `99999999999999999999999999999999999999999999999`. I think we can remove this sentry log, and just add a code comment explaining

Reply:
Thanks @gigxz! Looks like it's due to text_searcher trying to match by db pk, which is a 4 byte signed int, so any value greater than  2147483647 would trigger this exception. This seems a little funky. If an external ID exceeded this number it would be unsearchable.

I wonder if we need to search by Client.id at all? Internal db PKs aren't exposed to users afaik. @eanders  is that true?

Seems like could just make the following change to text_searcher and then remove all the range error rescues. 
```diff
-  where = sa[:PersonalID].eq(text).or(sa[:id].eq(text))
+ where = sa[:PersonalID].eq(text)
```

Reply:
So, we use primary Client.id keys all the time, so yeah, these need to be searchable.  It's too bad they are only Int4.  I thought Rails usually made primary keys Int8, but something/someone early in the warehouse development set them to be Int4s on most tables.  We've fixed a bunch of other keys, but this one is pretty pervasive.  

Can we just rescue the exception and search the other items if we hit it?

Reply:
Oh okay, good to know, thanks! In that case I think we can omit the the PK condition if the term would exceed the max for the column. I'll make that change. Right now it's all-or nothing